src/test/java/net/snowflake/ingest/streaming/internal/datatypes/AbstractDataTypeTest.java

```diff
@@ -575,17 +595,56 @@ protected void testIcebergIngestAndQuery(
     for (Object expectedValue : expectedValues) {
       Assertions.assertThat(resultSet.next()).isTrue();
       Object res = resultSet.getObject(1);
-      if (expectedValue instanceof BigDecimal) {
-        Assertions.assertThat(res)
-            .usingComparatorForType(BigDecimal::compareTo, BigDecimal.class)
-            .usingRecursiveComparison()
-            .isEqualTo(expectedValue);
-      } else if (expectedValue instanceof Timestamp) {
-        Assertions.assertThat(res.toString()).isEqualTo(expectedValue.toString());
-      } else {
-        Assertions.assertThat(res).isEqualTo(expectedValue);
-      }
+      assertEqualValues(expectedValue, res);
     }
     Assertions.assertThat(resultSet.next()).isFalse();
   }
+
+  protected void verifyMultipleColumns(
+      String tableName,
+      List<Map<String, Object>> values,
+      List<Map<String, Object>> expectedValues,
+      String orderBy)
+      throws Exception {
+    SnowflakeStreamingIngestChannel channel = openChannel(tableName);
+    Set<String> keySet = new HashSet<>();
+    for (Map<String, Object> value : values) {
+      String offsetToken = UUID.randomUUID().toString();
+      channel.insertRow(value, offsetToken);
+      TestUtils.waitForOffset(channel, offsetToken);
```

Comment:
why not do a single waitForOffset after the for loop ends?

Reply:
Moved to the end of for loop.